acra-censor/handlers/blacklist_handler.go

```diff
@@ -273,101 +267,78 @@ func checkSinglePatternMatch(queryNodes []sqlparser.SQLNode, patternNodes []sqlp
 }
 
 //handle %%SELECT%% pattern
-func handleSelectPattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
-	var emptySelect *sqlparser.Select
-	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
-		if strings.EqualFold(sqlparser.String(patternNodes[0].(*sqlparser.Select).SelectExprs), SelectConfigPlaceholderReplacerPart2) {
-			return true, true
-		} else {
-			return true, false
+func handleSelectPattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
+	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(patternNodes[0]) {
+		if patternNodeSelect, ok := patternNodes[0].(*sqlparser.Select); ok && strings.EqualFold(sqlparser.String(patternNodeSelect.SelectExprs), SelectConfigPlaceholderReplacerPart2) {
+			return true
 		}
 	}
-	return false, false
+	return false
 }
 
 //handle SELECT %%COLUMN%% .. %%COLUMN%% pattern
-func handleSelectColumnPattern(queryNodes, patternNodes []sqlparser.SQLNode) (patternDetected bool, matchOccurred bool) {
-	var emptySelect *sqlparser.Select
-	if reflect.TypeOf(queryNodes[0]) == reflect.TypeOf(emptySelect) && reflect.TypeOf(patternNodes[0]) == reflect.TypeOf(emptySelect) {
-		if strings.Contains(sqlparser.String(patternNodes[0].(*sqlparser.Select).SelectExprs), ColumnConfigPlaceholderReplacer) {
-			if len(patternNodes[0].(*sqlparser.Select).SelectExprs) == len(queryNodes[0].(*sqlparser.Select).SelectExprs) {
-				return true, true
+func handleSelectColumnPattern(queryNodes, patternNodes []sqlparser.SQLNode) bool {
+	matchDetected := false
+	if len(patternNodes) != len(queryNodes) {
+		return false
+	}
+	for index, patternNode := range patternNodes {
+		if index == 0 || reflect.DeepEqual(patternNode, queryNodes[index]) {
+			continue
+		}
+		if patternNodeColName, ok := patternNode.(*sqlparser.ColName); ok && patternNodeColName != nil {
```

Comment:
what if type of node in query != type of node in pattern and it's not ColName? Can we check that they both has same type and then check value of `.Name`?